toolkit/tools/scheduler/schedulerutils/buildworker.go

```diff
@@ -186,17 +189,66 @@ func getBuildDependencies(node *pkggraph.PkgNode, pkgGraph *pkggraph.PkgGraph, g
 	return
 }
 
+func parseCheckSection(logFile string) (err error) {
+	file, err := os.Open(logFile)
+	// If we can't open the log file, that's a build error.
+	if err != nil {
+		logger.Log.Errorf("Failed to open log file '%s' while checking package test results. Error: %v", logFile, err)
+		return
+	}
+	defer file.Close()
+	for scanner := bufio.NewScanner(file); scanner.Scan(); {
+		currLine := scanner.Text()
+		// Anything besides 0 is a failed test
+		if strings.Contains(currLine, "CHECK DONE") && !strings.Contains(currLine, "EXIT STATUS 0") {
+			failedLogFile := strings.TrimSuffix(logFile, ".log")
+			failedLogFile = fmt.Sprintf("%s-FAILED_TEST-%d.log", failedLogFile, time.Now().UnixMilli())
+			err = os.Rename(logFile, failedLogFile)
+			if err != nil {
+				logger.Log.Errorf("Log file rename failed. Error: %v", err)
+				return
+			}
+			err = fmt.Errorf("package test failed. Test status line: %s", currLine)
+			return
+		} else if strings.Contains(currLine, "CHECK DONE") {
+			return
+		}
```

Comment:
Simplification idea, since we're calling `strings.Contains(currLine, "CHECK DONE")` twice:
```suggestion
		if strings.Contains(currLine, "CHECK DONE") {
			if strings.Contains(currLine, "EXIT STATUS 0") {
				return
			}

			failedLogFile := strings.TrimSuffix(logFile, ".log")
			failedLogFile = fmt.Sprintf("%s-FAILED_TEST-%d.log", failedLogFile, time.Now().UnixMilli())
			err = os.Rename(logFile, failedLogFile)
			if err != nil {
				logger.Log.Errorf("Log file rename failed. Error: %v", err)
				return
			}
			err = fmt.Errorf("package test failed. Test status line: %s", currLine)
			return
		}
```